test/puppetlabs/puppetdb/scf/migrate_test.clj

```diff
@@ -242,3 +243,15 @@
     (is (thrown-with-msg? IllegalStateException
                           #"Found an old and unuspported database migration.*"
                           (migrate! *db*)))))
+
+(deftest test-upgrade-migration
```

Comment:
I could just remove the comment entirely if it's confusing. The issue is when there a row in the migration table from 1.x or 2.x, the upgrade will fail. This test covers that by adding a row for each prior migration (all of 1.x and 2.x)